### ISSUE TITLE
NonBlockingJsonParser requires trailing whitespace for numbers, boolean, and null

### ISSUE THREAD

**Issue by xma000** [FasterXML/jackson-core#454]
When I use `NonBlockingJsonParser` to parse numbers, `true`, `false`, and `null`, I must add a trailing whitespace. Otherwise, the next token is `NOT_AVAILABLE`. String, object and array are fine. The following code demonstrates the problem.

```java
    byte[] data = "true".getBytes();
    NonBlockingJsonParser parser = (NonBlockingJsonParser) new JsonFactory().createNonBlockingByteArrayParser();
    parser.feedInput(data, 0, data.length);
    assertEquals(parser.nextToken(), JsonToken.VALUE_TRUE);
```

**Comment by cowtowncoder** [comment]
This is by design, i.e. feature not bug. Otherwise it would be impossible to distinguish case of only getting part of the token (like "13" of "1356"); or, in case of `true` and `false`, invalid token like `trueenough`.

But note that you do not have to provide trailing space (although that works) -- instead, if this is end of input, that must be indicated.

**Comment by cowtowncoder** [comment]
@xma000 no problem -- it is a valid question, and something I should probably document better.

Also: if you have other challenges or anything, will be interested in hearing. I am really looking forward to non-blocking parsing getting more usage. It is new feature, but potentially very useful, and I want it to become stable and easy/easier to use.

**Comment by xma000** [comment]
Non-blocking parser enables us to implement parsers for [reactive streams](http://www.reactive-streams.org/). It works pretty well for me so far.

For a minor annoyance, `JsonFactory.createNonBlockingByteArrayParser` method could return a `NonBlockingJsonParser` so that I don't have to cast it.
